### PR TITLE
Create a method to consistently draw boxes in OrbitGl

### PR DESCRIPTION
**Pull request by karupayun**
In this PR we are just moving a method to draw boxes from SchedulerTrack to TimelineInfoInterface. 
This will help to solve the glitches when drawing boxes in the Capture Window (for example, see 
https://github.com/google/orbit/pull/4127 for ThreadTracks). The implementation of this method
will extend the box to the pixel borders as it's doing in the SchedulerTrack.  

This method is needed for ThreadStates (https://github.com/google/orbit/pull/4274), to draw
consistently after applying the pixel rendering optimization.

Test: Load a capture, everything looks as expected.

### CODE REVIEW

File: src/OrbitGl/ThreadStateBar.cpp [hunk 1]
```diff
@@ -297,12 +297,10 @@ void ThreadStateBar::DoUpdatePrimitives(PrimitiveAssembler& primitive_assembler,
...
 
-        const float x0 = timeline_info_->GetWorldFromTick(slice.begin_timestamp_ns());
-        const float x1 = timeline_info_->GetWorldFromTick(slice.end_timestamp_ns());
-        const float width = x1 - x0;
-
-        const Vec2 pos{x0, GetPos()[1]};
-        const Vec2 size{width, GetHeight()};
+        auto [box_start_x, box_width] = timeline_info_->GetBoxPosXAndWidthFromTicks(
+            slice.begin_timestamp_ns(), slice.end_timestamp_ns());
+        const Vec2 pos{box_start_x, GetPos()[1]};
```
Comment (florian-kuebler) [inline src/OrbitGl/ThreadStateBar.cpp:302]: Nit: Can you add the new line as it was before?

**Review by florian-kuebler — APPROVED** [review]
LGTM

**Review by reichlfl — APPROVED** [review]
I'm fine with this, but I want to point out that I think a better design would be to not have `GetBoxPosXAndWidthFromTicks` be part of the `TimelineInfoInterface`. You should strife to keep interfaces minimal, and the implementation of `GetBoxPosXAndWidthFromTicks` only relies on public functionality already available through the interface, so you could also make it a free function that receives a `TimelineInfoInterface` as an additional parameter. This way, you also wouldn't need the implementation in the Mock.

**Comment by karupayun** [comment]
> I'm fine with this, but I want to point out that I think a better design would be to not have `GetBoxPosXAndWidthFromTicks` be part of the `TimelineInfoInterface`. You should strife to keep interfaces minimal, and the implementation of `GetBoxPosXAndWidthFromTicks` only relies on public functionality already available through the interface, so you could also make it a free function that receives a `TimelineInfoInterface` as an additional parameter. This way, you also wouldn't need the implementation in the Mock.

As discussed offline, I fully agree. TimelineInfoUtils makes a lot of sense for a place to have this function. As it's only one function, and you've already approved, I will merge it as it is. Thanks for the input anyway!

**Pull request merged**